src/common/CvtFormat.cpp

```diff
@@ -0,0 +1,1678 @@
+						status_exception::raise(Arg::Gds(isc_invalid_timezone_region) << string(str + strOffset, parsedTimezoneNameLength));
+
+					strOffset += parsedTimezoneNameLength;
+					break;
+				}
+				default:
+					invalidPatternException(patternStr, cb);
+			}
+		}
+
+		processStringToDateTimeDeferredTokens(deferredTokens, outCvtData, cb);
+
+		for (; strOffset < strLength; strOffset++)
+		{
+			if (!isSeparator(str[strOffset]))
+				break;
+		}
+		if (strOffset < strLength)
+			cb->err(Arg::Gds(isc_trailing_part_of_string) << string(str + strOffset));
+	}
+
+	void timeStampToUtc(ISC_TIMESTAMP_TZ& timestampTZ, USHORT sessionTimeZone, EXPECT_DATETIME expectedType,
+		Callbacks* cb)
+	{
+		if (expectedType == expect_sql_time_tz || expectedType == expect_timestamp_tz || timestampTZ.time_zone != sessionTimeZone)
+			TimeZoneUtil::localTimeStampToUtc(timestampTZ);
+
+		if (timestampTZ.time_zone != sessionTimeZone)
+		{
+			if (expectedType == expect_sql_time)
+			{
+				ISC_TIME_TZ timeTz;
+				timeTz.utc_time = timestampTZ.utc_timestamp.timestamp_time;
+				timeTz.time_zone = timestampTZ.time_zone;
+				timestampTZ.utc_timestamp.timestamp_time = TimeZoneUtil::timeTzToTime(timeTz, cb);
+			}
+			else if (expectedType == expect_timestamp)
+				*(ISC_TIMESTAMP*) &timestampTZ = TimeZoneUtil::timeStampTzToTimeStamp(timestampTZ, sessionTimeZone);
+		}
+	}
+
+	void validateTimeStamp(const ISC_TIMESTAMP timestamp, const EXPECT_DATETIME expectedType, const dsc* desc,
+		Callbacks* cb)
+	{
+		if (!NoThrowTimeStamp::isValidTimeStamp(timestamp))
+		{
+			switch (expectedType)
+			{
+				case expect_sql_date:
+					cb->err(Arg::Gds(isc_date_range_exceeded));
+					break;
+				case expect_sql_time:
+				case expect_sql_time_tz:
+					cb->err(Arg::Gds(isc_time_range_exceeded));
+					break;
+				case expect_timestamp:
+				case expect_timestamp_tz:
+					cb->err(Arg::Gds(isc_datetime_range_exceeded));
+					break;
+				default: // this should never happen!
+					CVT_conversion_error(desc, cb->err);
+					break;
+			}
+		}
+	}
+
+	ISC_TIMESTAMP_TZ constructTimeStampTz(const StringToDateTimeData& cvtData, Callbacks* cb)
+	{
+		ISC_TIMESTAMP_TZ timestampTZ;
+		timestampTZ.utc_timestamp = NoThrowTimeStamp::encode_timestamp(&cvtData.times, cvtData.fractions);
+
+		const ISC_USHORT sessionTimeZone = cb->getSessionTimeZone();
+
+		if (cvtData.timezoneOffsetInMinutes == UninitializedTimezoneOffsetValue && cvtData.timezoneId == TimeZoneTrie::UninitializedTimezoneId)
+			timestampTZ.time_zone = sessionTimeZone;
+		else if (cvtData.timezoneId != TimeZoneTrie::UninitializedTimezoneId)
+			timestampTZ.time_zone = cvtData.timezoneId;
+		else
+		{
+			timestampTZ.time_zone = TimeZoneUtil::makeFromOffset(sign(cvtData.timezoneOffsetInMinutes),
+				abs(cvtData.timezoneOffsetInMinutes) / 60, abs(cvtData.timezoneOffsetInMinutes) % 60);
+		}
+
+		return timestampTZ;
+	}
+}
+
+string CVT_datetime_to_format_string(const dsc* desc, const string& format, Callbacks* cb)
```

Comment:
It looks a bit weird that these public functions are declared in `cvt.h` and `CvtFormats.h` is not a header to be included by users of this new "module". I would instead rename current 'CvtFormat.h` to `CvtFormatImpl.h` and declare the public functions in `CvtFormat.h` (with `CVT_format_` prefix).